kvbc/src/merkle_tree_db_adapter.cpp

```diff
@@ -354,7 +367,10 @@ BatchedInternalNode DBAdapter::Reader::get_internal(const InternalNodeKey &key)
 
 BlockId DBAdapter::addBlock(const SetOfKeyValuePairs &updates, const OrderedKeysSet &deletes) {
   const auto blockId = getLastReachableBlockId() + 1;
-  const auto status = db_->multiPut(lastReachableBlockDbUpdates(updates, deletes, blockId));
+  int64_t data_raw_size = 0;
+  auto data = lastReachableBlockDbUpdates(updates, deletes, blockId, data_raw_size);
+  writeCommitSizeSummary_.Get().Observe(data_raw_size);
```

Comment:
Why do we need to calculate data_raw_size in many different places instead of doing it here from the resulting lastReachableBlockDbUpdates SetOfKeyValuePairs?

Reply:
I agree with @yuliasherman 's comment - we can just use the `dbUpdates` return variable in `lastReachableBlockDbUpdates()`. It will also cover the case of replica receiving a block over State Transfer.

Reply:
I would prefer to avoid rotating over the updates again. This may harm performance, isn't it?

Reply:
Don't think so. It's a quick cpu operation.

Reply:
And what if the size of the update map is very big?
Isn't it better to make only a single loop over it?

Reply:
You can measure how long it takes for 1000 keys :-)